src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/FileBasedPrograms/FileBasedProgramsProjectSystem.cs

```diff
@@ -107,6 +169,21 @@ TextDocument AddPrimordialDocument(DocumentUri uri, SourceText documentText, str
     public async ValueTask<bool> TryRemoveMiscellaneousDocumentAsync(DocumentUri uri)
     {
         var documentPath = GetDocumentFilePath(uri);
+
+        // First try to remove from the canonical project
+        using (await _canonicalProjectGate.DisposableWaitAsync(CancellationToken.None))
+        {
+            if (_canonicalMiscFilesProject != null)
+            {
+                var removedFromCanonical = await _canonicalMiscFilesProject.RemoveDocumentAsync(documentPath, CancellationToken.None);
```

Comment:
This will also be unnecessary when the change is made to use derived Workspaces Project instances, instead of adding documents to the LoadedProject.